msys2-runtime/0024-path.cc-Allow-native-unprivileged-symlink.patch

```diff
@@ -0,0 +1,28 @@
+From 07dc2f408e5da9615c2eb6db07979a1a7592ef78 Mon Sep 17 00:00:00 2001
+From: Einbert_Xeride <einbert-xeride@outlook.com>
+Date: Tue, 25 Apr 2017 02:39:07 +0800
+Subject: [PATCH 24/24] path.cc: Allow native unprivileged symlink
+
+---
+ winsup/cygwin/path.cc | 5 +++--
+ 1 file changed, 3 insertions(+), 2 deletions(-)
+
+diff --git a/winsup/cygwin/path.cc b/winsup/cygwin/path.cc
+index 13f4e39f5..e3aaae3f6 100644
+--- a/winsup/cygwin/path.cc
++++ b/winsup/cygwin/path.cc
+@@ -1813,8 +1813,9 @@ symlink_native (const char *oldpath, path_conv &win32_newpath)
+     }
+   /* Try to create native symlink. */
+   if (!CreateSymbolicLinkW (final_newpath->Buffer, final_oldpath->Buffer,
+-			    win32_oldpath.isdir ()
+-			    ? SYMBOLIC_LINK_FLAG_DIRECTORY : 0))
++				(win32_oldpath.isdir ()
++				? SYMBOLIC_LINK_FLAG_DIRECTORY : 0)
++				| 0x2 /* SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE */ ))
```

Comment:
Are you sure there is works on both Win 7 and Win 10 for directories?
If `win32_oldpath.isdir()` is true you will get `0x1 | 0x2` which results in 0x3.